sbin/geom/class/eli/geom_eli.c

```diff
@@ -1724,3 +1736,52 @@ eli_dump(struct gctl_req *req)
 		printf("\n");
 	}
 }
+
+static void
+eli_verify(struct gctl_req *req)
+{
+	struct g_eli_metadata md;
+	const char *prov;
+	int nargs;
+	unsigned char key[G_ELI_USERKEYLEN], mkey[G_ELI_DATAIVKEYLEN];
+	unsigned int nkey;
+	int error;
+
+	nargs = gctl_get_int(req, "nargs");
+	if (nargs != 1) {
+		gctl_error(req, "Invalid number of arguments.");
+		return;
+	}
+	prov = gctl_get_ascii(req, "arg0");
+
+	if (eli_metadata_read(req, prov, &md) == -1)
+		return;
+
+	if (md.md_keys == 0) {
```

Comment:
This code is mostly a copy of eli_setkey_detached.  Could you please extract the common portion out instead?